test/built-ins/Promise/prototype/finally/subclass-reject-count.js

```diff
@@ -16,7 +16,8 @@ class FooPromise extends Promise {
   }
 }
 
-FooPromise.reject().finally(() => {}).then($ERROR).catch(() => {
-  assert.sameValue(7, count);
-  $DONE();
-});
+FooPromise.reject().finally(() => {}).then(value => {
+  throw new Test262Error("Expected Promise to be rejected, got: resolved with " + value);
+}, () => {
+  assert.sameValue(count, 7);
+}).then($DONE, $DONE);
```

Comment:
This is interesting, it seems like in the unchanged version we would not capture the `$DONE` for the async execution of the test properly. If the assertion failed at line 20 (`count === 7`) we would never reach $DONE and therefore we'd need to wait for a timeout. This format captures the end correctly if the test fails or pass.